web/src/main/java/org/springframework/security/web/server/header/CrossOriginPoliciesServerHttpHeadersWriter.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.header;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * Inserts cross-origin policies headers. Specifically it can add the following headers:
+ * <ul>
+ * <li>Cross-Origin-Embedder-Policy</li>
+ * <li>Cross-Origin-Opener-Policy</li>
+ * <li>Cross-Origin-Resource-Policy</li>
+ * </ul>
+ *
+ * @author Marcus Da Coregio
+ * @since 5.6
+ * @see <a href=
+ * "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy">
+ * Cross-Origin-Opener-Policy</a>
+ * @see <a href=
+ * "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy">
+ * Cross-Origin-Embedder-Policy</a>
+ * @see <a href=
+ * "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy">
+ * Cross-Origin-Resource-Policy</a>
+ */
+public final class CrossOriginPoliciesServerHttpHeadersWriter implements ServerHttpHeadersWriter {
```

Comment:
The same feedback as `CrossOriginPoliciesHeadersWriter.java` applies here, about splitting this into 3 separate `ServerHttpHeadersWriter`s.